pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/server/ServerPlanRequestUtils.java

```diff
@@ -147,6 +147,13 @@ private static ServerPlanRequestContext build(OpChainExecutionContext executionC
     // 2. set pinot query options according to requestMetadataMap
     updateQueryOptions(pinotQuery, executionContext);
 
+    if (QueryOptionsUtils.isNullHandlingEnabled(pinotQuery.getQueryOptions())
```

Comment:
i am not fully sure i understand the logic here. but I think it is OK to move this to a separate PR (not going to change the behavior if something is broken; and whatever works still works)

Reply:
The logic here is:
```
If a query is executed with V2 engine
  And is executed with null handling
  And any table consumed by the query is not using column level nullability
Then abort the execution with an error saying column null handling must be enable for all tables involved in a multi-stage query with null handling
```

I've been asked several times about queries returning strange results in V2 when using null handling. We don't support that but users don't know it. Therefore I strongly suggest to fail fast in order to be explicit about this limitation. This will also help users to understand that is important to migrate to column level nullability.

Reply:
But I'm ok removing this from the PR give it side affects several tests which are not actually testing nullability

Reply:
Removed in ac5c5e51c8ce8d418cc7cf35590e1a4e5a273a55